table/block_based/partitioned_filter_block.cc

```diff
@@ -154,7 +153,14 @@ Slice PartitionedFilterBlockBuilder::Finish(
     // indicate we expect more calls to Finish
     *status = Status::Incomplete();
     finishing_filters = true;
-    return filters.front().filter;
+
+    last_filter_entry_key = filters.front().key;
+    Slice filter = filters.front().filter;
+    if (filter_data != nullptr) {
+      *filter_data = std::move(filters.front().filter_data);
+    }
+    filters.pop_front();
```

Comment:
If `filter_data == nullptr` then the `Slice filter` will now refer to freed data. So we either need to require `filter_data != nullptr` at least for partitioned filters or have another field we can save to (`last_filter_data`?) to avoid or solve this case.

Reply:
I lean toward the second idea a bit better because it is consistent with how full filter handles this case :) will update on this soon